suite-native/device-manager/src/components/DevicesToggleButton.tsx

```diff
@@ -8,19 +11,45 @@ type DevicesToggleButtonProps = {
     onDeviceButtonTap: () => void;
 };
 
+const CHEVRON_ANIMATION_DURATION = 300;
+
 export const DevicesToggleButton = ({
     deviceButtonState = 'closed',
     onDeviceButtonTap,
-}: DevicesToggleButtonProps) => (
-    <Button
-        size="small"
-        colorScheme="tertiaryElevation0"
-        viewRight={deviceButtonState === 'open' ? 'chevronUp' : 'chevronDown'}
-        onPress={onDeviceButtonTap}
-        testID="@device-manager/devices/toggle"
-    >
-        <Text variant="hint">
-            <Translation id="deviceManager.deviceButtons.devices" />
-        </Text>
-    </Button>
-);
+}: DevicesToggleButtonProps) => {
+    const isOpen = useSharedValue(false);
+
+    const animatedChevronStyle = useAnimatedStyle(() => ({
+        transform: [
+            {
+                rotate: withTiming(`${isOpen.value ? -180 : 0}deg`, {
+                    duration: CHEVRON_ANIMATION_DURATION,
+                }),
+            },
+        ],
+    }));
+
+    useEffect(() => {
+        'worklet';
+
+        isOpen.value = deviceButtonState === 'open';
+    }, [deviceButtonState, isOpen]);
```

Comment:
Can you reuse the `deviceButtonState` instead of creating new value using `useSharedValue`? 

If not, I'd prefer to extend the functionality of the `onDeviceButtonTap` by creating and extending function which would handle this functionality. No reason to add a `useEffect` for this IMO. 

Something like this maybe:
```
const handleOnDeviceButtonTap = () => {
    onDeviceButtonTap()
    // Other things you want to add to this function and need to be in this component
}
```

You might also consider sending the style directly to this component as a prop. However I think the useEffect here is not very pretty.

Reply:
I agree with @juriczech. The `isOpen` and `useEffect` are redundant here. You can access the `deviceButtonState` directly in the `useAnimatedStyle`.

Reply:
8f6bff1956